tensorzero-internal/src/config_parser.rs

```diff
@@ -52,6 +53,27 @@ pub struct GatewayConfig {
     pub enable_template_filesystem_access: bool,
 }
 
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub struct GCPProviderTypeConfig {
+    #[serde(default)]
+    pub batch: Option<GCPBatchConfigType>,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(tag = "storage_type", rename_all = "snake_case")]
+pub enum GCPBatchConfigType {
+    None,
```

Comment:
why do we have None here? this seems like a redundant state in the type system and we can get by with None in `GCPProviderTypeConfig.batch`

Reply:
When we add per-model-provider batch configs, this will allow overwriting the `provider-types` config to `None`.